<a href="https://colab.research.google.com/github/abdullahkrs/e-Portfolio-Machine-Learning-January-2025/blob/main/Multiple_Linear_Regression.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


# 📊 Multiple Linear Regression with cars.csv

This notebook demonstrates **Multiple Linear Regression** using the `cars.csv` dataset. The goal is to predict **CO2 emissions** based on both a car’s **weight** and **engine volume**.

## 📌 Description
- **Features (X)**: `Weight`, `Volume`
- **Target (y)**: `CO2`
- **Algorithm**: `LinearRegression` from `scikit-learn`

## 🔍 Objective
To understand how multiple factors together influence a target variable, and how to interpret model coefficients in a multivariate regression setup.

## 🧪 Learning Outcomes
- Upload and preprocess real-world data
- Apply and interpret multiple linear regression
- Understand the individual impact of each feature on the prediction

---


In [1]:

import pandas as pd
from sklearn.linear_model import LinearRegression
from google.colab import files
import io

# Upload CSV
uploaded = files.upload()
df = pd.read_csv(io.BytesIO(uploaded[list(uploaded.keys())[0]]))

# Features and target
X = df[['Weight', 'Volume']]
y = df['CO2']

# Fit model
model = LinearRegression()
model.fit(X, y)

# Display coefficients
print(f"Intercept: {model.intercept_:.2f}")
print(f"Weight Coefficient: {model.coef_[0]:.4f}")
print(f"Volume Coefficient: {model.coef_[1]:.4f}")

# Predict sample case
predictedCO2 = model.predict([[2300, 1300]])
print(f"Predicted CO2 for Weight=2300kg, Volume=1300cc: {predictedCO2[0]:.2f}")


Saving cars (1).csv to cars (1).csv
Intercept: 79.69
Weight Coefficient: 0.0076
Volume Coefficient: 0.0078
Predicted CO2 for Weight=2300kg, Volume=1300cc: 107.21


/usr/local/lib/python3.11/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Coefficient

The coefficient is a factor that describes the relationship with an unknown variable.
In this case, we can ask for the coefficient value of weight against CO2, and for volume against CO2. The answer(s) we get tells us what would happen if we increase, or decrease, one of the independent values.

In [3]:
print(model.coef_)

[0.00755095 0.00780526]


The result array represents the coefficient values of weight and volume.

Weight: 0.00755095
Volume: 0.00780526

These values tell us that if the weight increase by 1kg, the CO2 emission increases by 0.00755095g.

And if the engine size (Volume) increases by 1 cm3, the CO2 emission increases by 0.00780526 g.

I think that is a fair guess, but let test it!

We have already predicted that if a car with a 1300cm3 engine weighs 2300kg, the CO2 emission will be approximately 107g.

What if we increase the weight with 1000kg (from 2300 to 3300) what will be the CO2 emission?

Ans: 107.2087328 + (1000 * 0.00755095) = 114.75968

In [6]:
predictedCO2 = model.predict([[3300, 1300]])

print(f"Predicted CO2 for Weight=3300kg, Volume=1300cc: {predictedCO2[0]:.2f}")


Predicted CO2 for Weight=3300kg, Volume=1300cc: 114.76


/usr/local/lib/python3.11/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(
